source/rules/roles.txt

```diff
@@ -511,3 +518,16 @@ deploying configuration files with {+cli+}:
          .. code-block:: bash
             
             {+cli-bin+} push
+
+.. note:: Security Consideration for App Services Role-based Permissions
+
+  While Role-based Permissions and :ref:`Filters <filters>` can hide specific 
+  documents and fields within a collection there is a potential that 
+  data can be exposed if the system allows arbitrary 
+  queries to access the collection. For example, queries or functions that 
+  raise errors depending on the values stored in a collection (such 
+  as division-by-zero errors) may reveal information about documents, even if 
+  a role or filter prevents the querying user from viewing 
+  documents directly. Users may also make inferences about the underlying data 
+  in other ways (such as by measuring query execution time, which can be affected 
+  by the data's distribution). 
```

Comment:
In terms of what's in the ticket this covers everything. However, I think it may be a bit frustrating for devs to read this and not really have anything actionable.

Maybe we can add a final sentence that's basically like "be aware that this is possible and audit your data access patterns if it's a concern for you"